recipes/thrust/all/conanfile.py

```diff
@@ -0,0 +1,39 @@
+from conans import ConanFile, tools
+import os
+
+class ThrustConan(ConanFile):
+    name = "thrust"
+    license = "Apache-2.0"
+    description = ("Thrust is a parallel algorithms library which resembles"
+                   "the C++ Standard Template Library (STL).")
+    topics = ("parallel", "stl", "header-only")
+    homepage = "https://thrust.github.io/"
+    url = "https://github.com/conan-io/conan-center-index"
+    no_copy_source = True
+    options = {"thrust_device_system": ["cuda", "cpp", "omp", "tbb"]}
+    default_options = {"thrust_device_system": "cuda"}
```

Comment:
Should we add the `requires(xxxx)` to `cuda`, `omp`, `tbb`,... depending on the value of the option? Even if it is a header-only, I'd need them in the graph if I want to use this package with the option activated.

I know we don't have packages for some of them, so my recommendation would be:
 * if `tbb`, then add the proper requires
 * for `cuda`, `omp`, write a `self.output.warn("Conan package for xxxx is not available, this package will use xxxx from the system")`
 * for `cpp` do nothing (I assume this is using native-stl implementation)

I would vote for using `tbb` in the test_package as it will check it is using the dependency, but it's ok for me with `cpp` too.

wdyt?

Reply:
I like it.